client/my-sites/plans/jetpack-plans/storage-tier-upgrade/hooks/use-available-storage-upgrade-products.ts

```diff
@@ -50,10 +50,14 @@ const useAvailableStorageUpgradeProducts = (
 			! allDurationsSubscriptions.includes( productSlug );
 	}, [ purchasedStorageSlugs ] );
 
-	return [
-		...TIER_1_SLUGS.map( getStorageUpgradeProduct ),
-		...TIER_2_SLUGS.map( getStorageUpgradeProduct ),
-	]
+	const allUpgradeProducts: SelectorProduct[] = [ ...TIER_1_SLUGS, ...TIER_2_SLUGS ].reduce(
+		( acc: SelectorProduct[], slug: string ) => {
+			return [ ...acc, ...( getStorageUpgradeProducts( slug ) as SelectorProduct[] ) ];
```

Comment:
Just a very small notice that we probably could use a `concat` here as it's more performant in this scenario.